src/circuit/transport.ts

```diff
@@ -90,102 +128,122 @@ export class Circuit implements Transport, Startable {
   }
 
   get [Symbol.toStringTag] () {
-    return 'libp2p/circuit-relay-v1'
+    return 'libp2p/circuit-relay-v2'
   }
 
-  async _onProtocol (data: IncomingStreamData) {
+  getPeerConnection (dstPeer: PeerId): Connection | undefined {
+    return this.components.connectionManager.getConnections(dstPeer)[0] ?? undefined
+  }
+
+  async _onProtocolV1 (data: IncomingStreamData) {
```

Comment:
@mpetrunic Should we disable circuit v1 relaying entirely? There seems to be no reason to register this handler as it will not relay.

Reply:
Yeah, sure! I was just copying logic that go-libp2p had but I'm pretty sure nowadays, there is no need for that!